id: PYSEC-0000-CVE-2022-21737
details: Tensorflow is an Open Source Machine Learning Framework. The implementation
  of `*Bincount` operations allows malicious users to cause denial of service by passing
  in arguments which would trigger a `CHECK`-fail. There are several conditions that
  the input arguments must satisfy. Some are not caught during shape inference and
  others are not caught during kernel implementation. This results in `CHECK` failures
  later when the output tensors get allocated. The fix will be included in TensorFlow
  2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3,
  and TensorFlow 2.5.3, as these are also affected and still in supported range.
affected:
- package:
    name: tensorflow-gpu
    ecosystem: PyPI
    purl: pkg:pypi/tensorflow-gpu
  ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    events:
    - introduced: "0"
    - fixed: 7019ce4f68925fd01cdafde26f8d8c938f47e6f9
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 2.5.3
    - introduced: 2.6.0
    - fixed: 2.6.3
references:
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/7019ce4f68925fd01cdafde26f8d8c938f47e6f9
- type: WEB
  url: https://github.com/tensorflow/tensorflow/blob/5100e359aef5c8021f2e71c7b986420b85ce7b3d/tensorflow/core/kernels/bincount_op.cc
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f2vv-v9cg-qhh7
aliases:
- CVE-2022-21737
- GHSA-f2vv-v9cg-qhh7
modified: "2022-02-09T05:08:00Z"
published: "2022-02-03T14:15:00Z"
